kolibri/plugins/coach/assets/src/state/actions/exam.js

```diff
@@ -271,25 +271,67 @@ function showExamReportDetailPage(store, classId, examId) {
   store.dispatch('CORE_SET_PAGE_LOADING', false);
 }
 
-function activateExam() {
-  this.displayModal(false);
+function activateExam(store, examId) {
+  ExamResource.getModel(examId).save({ active: true }).then(
+    () => {
+      const exams = store.state.pageState.exams;
+      const examIndex = exams.findIndex(exam => exam.id === examId);
+      exams[examIndex].active = true;
+
+      store.dispatch('SET_EXAMS', exams);
+      this.displayModal(false);
+    },
+    error => CoreActions.handleError(store, error)
+  );
 }
 
-function deactivateExam() {
-  this.displayModal(false);
+function deactivateExam(store, examId) {
+  ExamResource.getModel(examId).save({ active: false }).then(
+    () => {
+      const exams = store.state.pageState.exams;
+      const examIndex = exams.findIndex(exam => exam.id === examId);
```

Comment:
Since we seem to be doing a couple of operations like this, do you think it'd be helpful to reformat the `exams` array to be an object? Something like
```
{
  examId: {examprops},
}
```

seems to be a good way to do it. But I can see how this might not be ideal if some of your other logic relies on it being an exam

Otherwise, i think it'd be helpful to make this a method so that you don't have to keep rewriting it